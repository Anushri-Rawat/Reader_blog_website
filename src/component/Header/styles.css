.nav {
  min-height: 60px;
  box-shadow: 0 0 4px grey;
  padding: 0 3% !important;
}
