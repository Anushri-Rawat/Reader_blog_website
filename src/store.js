import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import posts from "./reducers";

const initialState = {
  loading: false,
  posts: JSON.parse(localStorage.getItem("posts")) || [],
};

const store = () =>
  createStore(posts, initialState, composeWithDevTools(applyMiddleware(thunk)));

export default store;
