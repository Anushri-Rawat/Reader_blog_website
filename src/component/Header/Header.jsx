import React from "react";
import "./styles.css";

const Header = () => {
  return (
    <div
      className="nav justify-content-between align-items-center"
      data-testid="header"
    >
      <h3>READER</h3>
      <div className="d-flex gap-2 gap-md-4 align-items-center">
        <i className="bi bi-bell"></i>
        <img
          src="https://th.bing.com/th/id/R.da2e546841da40cdcf60061743233500?rik=IeO7Sr%2fkUW54wQ&riu=http%3a%2f%2fwww.venmond.com%2fdemo%2fvendroid%2fimg%2favatar%2fbig.jpg&ehk=JihI5nQ0BOd0W%2bZVhtIWmqwac0NMyRMOV7%2bzryywg%2fg%3d&risl=&pid=ImgRaw&r=0"
          alt="user-img"
          className="rounded-circle"
          style={{ width: "40px", height: "40px" }}
        />
        <button
          className="btn btn-outline-secondary"
          data-bs-toggle="modal"
          data-bs-target="#addBlogForm"
        >
          <i className="bi bi-pencil-square"></i>
          <span style={{ marginLeft: "6px" }}>Add Blog</span>
        </button>
      </div>
    </div>
  );
};

export default Header;
