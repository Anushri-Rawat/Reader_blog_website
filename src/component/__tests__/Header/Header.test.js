import { render as rtlRender, screen, cleanup } from "@testing-library/react";
import Header from "../../Header/Header";
import store from "../../../store";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

const render = (component) =>
  rtlRender(<Provider store={store()}>{component}</Provider>);

test("should render header component", () => {
  render(<Header />);
  const headerElement = screen.getByTestId("header");
  expect(headerElement).toBeInTheDocument();
});

test("matches snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store()}>
        <Header />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
