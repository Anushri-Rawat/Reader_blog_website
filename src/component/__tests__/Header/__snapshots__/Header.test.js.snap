// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="nav justify-content-between align-items-center"
  data-testid="header"
>
  <h3>
    READER
  </h3>
  <div
    className="d-flex gap-2 gap-md-4 align-items-center"
  >
    <i
      className="bi bi-bell"
    />
    <img
      alt="user-img"
      className="rounded-circle"
      src="https://th.bing.com/th/id/R.da2e546841da40cdcf60061743233500?rik=IeO7Sr%2fkUW54wQ&riu=http%3a%2f%2fwww.venmond.com%2fdemo%2fvendroid%2fimg%2favatar%2fbig.jpg&ehk=JihI5nQ0BOd0W%2bZVhtIWmqwac0NMyRMOV7%2bzryywg%2fg%3d&risl=&pid=ImgRaw&r=0"
      style={
        Object {
          "height": "40px",
          "width": "40px",
        }
      }
    />
    <button
      className="btn btn-outline-secondary"
      data-bs-target="#addBlogForm"
      data-bs-toggle="modal"
    >
      <i
        className="bi bi-pencil-square"
      />
      <span
        style={
          Object {
            "marginLeft": "6px",
          }
        }
      >
        Add Blog
      </span>
    </button>
  </div>
</div>
`;
